### PR TITLE
Use apt-get to install Tanzu CLI plugin builder

### LINKED ISSUE #101: Use apt-get and tanzu builder install to install Tanzu CLI plugin builder instead of cloning and building from source code
**Is your feature request related to a problem? Please describe.**

When work on building CLI plugins started, the ability to download Tanzu CLI plugin builder using apt-get was not available. At that time, I decided to–as a temporary measure–git clone and build the Tanzu CLI plugin builder from source code.

**Describe the solution you'd like**

Now that apt-get is available, use that instead of building from source code.

**Describe alternatives you've considered**

None

**Additional context**

None

### PR DESCRIPTION
**Pull request by codegold79**
# Description
_Include a summary of the change. If this change fixes an issue, indicate that on the `Fixes` line. Briefly include
relevant context and motivation._

See issue #101.

Fixes #101

## Change Details
Check all that apply:
- [x] New feature 
- [ ] Bug fix 
- [ ] Non-code change 
- [ ] Non-breaking change 
- [ ] Breaking change
- [ ] Requires a documentation change

## Release Note

```release-note
Use apt-get instead of git clone to install Tanzu CLI plugin builder
```

# How Has This Been Tested?
_Describe any testing done to verify changes. Provide enough detail that tests can be reproduced._

# Checklist:
- [x] My code follows the [Contribution and Style guidelines](../../CONTRIBUTING.md) of this project
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added test details that prove my fix is effective or that my feature works

### CODE REVIEW

File: templates/Dockerfile [hunk 1]
```diff
@@ -67,17 +68,17 @@ ARG BUILDARCH
...
-    cd tanzu-cli/cmd/plugin/builder && \
-    go build -o /bin/tanzu-builder . && \
-    cd /workspace
+ARG TANZU_CLI_VERSION
+RUN apt-get update && \
+    apt-get install -y ca-certificates && \
+    printf "deb https://storage.googleapis.com/tanzu-cli-os-packages/apt tanzu-cli-jessie main" | tee /etc/apt/sources.list.d/tanzu.list && \
+    apt-get update --allow-insecure-repositories && \
+    apt-get install -y tanzu-cli=${TANZU_CLI_VERSION} --allow-unauthenticated && \
+    tanzu ceip set true && \
```
Comment (yharish991) [inline templates/Dockerfile:79]: any reason to set this to true?

File: templates/Dockerfile [hunk 2]
```diff
@@ -67,17 +68,17 @@ ARG BUILDARCH
...
-    git clone https://github.com/vmware-tanzu/tanzu-cli.git && \
-    cd tanzu-cli/cmd/plugin/builder && \
-    go build -o /bin/tanzu-builder . && \
-    cd /workspace
+ARG TANZU_CLI_VERSION
+RUN apt-get update && \
+    apt-get install -y ca-certificates && \
+    printf "deb https://storage.googleapis.com/tanzu-cli-os-packages/apt tanzu-cli-jessie main" | tee /etc/apt/sources.list.d/tanzu.list && \
+    apt-get update --allow-insecure-repositories && \
+    apt-get install -y tanzu-cli=${TANZU_CLI_VERSION} --allow-unauthenticated && \
```
[the remark this answers is not in the document]
Comment (codegold79) [reply templates/Dockerfile:78]: Yes. These flags are straight from CLI documentation: https://github.com/vmware-tanzu/tanzu-cli/blob/main/docs/quickstart/install.md#apt-debianubuntu

**Review by rajathagasthya — APPROVED** [review]

**Review by yharish991 — APPROVED** [review]

**Review by yharish991 — COMMENTED** [review]
Something for the future: The current approach installs tanzu cli for each plugin it is building, we need to optimize this just to install only once

**Comment by codegold79** [comment]
> Something for the future: The current approach installs tanzu cli for each plugin it is building, we need to optimize this just to install only once

Thanks, @yharish991 for the suggestion. I have made an issue so it's not forgotten. I'll try to remember to do that when we update the Tanzu CLI version (which will probably be soon): https://github.com/vmware-tanzu/build-tooling-for-integrations/issues/103

**Pull request merged**